app/core/exceptions.py

```
@@ -0,0 +1,282 @@
+        Parameters:
+            message (str): Human-readable error message; defaults to "Insufficient permissions".
+        """
+        super().__init__(
+            message=message, status_code=403, error_code="AUTHORIZATION_ERROR"
+        )
+
+
+class NotFoundError(APIError):
+    """Raised when a resource is not found"""
+
+    def __init__(self, resource: str, identifier: str = None):
```

Ruff (0.13.3)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)